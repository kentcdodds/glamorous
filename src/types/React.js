// @flow
import type {Component, Element} from 'react'

export type FunctionalComponent = (
  {[key: string]: any}, // props
  ?{[key: string]: any}, // context
) => Element<*> | null;

// eslint-disable-next-line no-undef
export type ComponentClass = Class<Component<*, *, *>>;

export type RefCallback<Instance> = (Instance | null) => void;
