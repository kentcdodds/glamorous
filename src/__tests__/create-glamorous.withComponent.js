/* eslint func-style:0, react/prop-types:0 */
import React from 'react'
import {render} from 'enzyme'
import * as jestGlamorReact from 'jest-glamor-react'
import glamorous from '../'

expect.extend(jestGlamorReact.matcher)
expect.addSnapshotSerializer(jestGlamorReact.serializer)

test('withComponent composes the component with provided styles', () => {
  const Text = glamorous.span({color: 'red', fontSize: 20})
  const View = Text.withComponent('div')
  expect(render(<View />)).toMatchSnapshotWithGlamor()
})

test('withComponent creates a new component with the provided tag', () => {
  const Text = glamorous.span({color: 'red', fontSize: 20})
  const View = Text.withComponent('div')
  expect(Text.comp).toBe('span')
  expect(View.comp).toBe('div')
})
