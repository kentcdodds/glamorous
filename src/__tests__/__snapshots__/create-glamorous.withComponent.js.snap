// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withComponent composes the component with styles 1`] = `
.css-5qt57y,
[data-css-5qt57y] {
  color: red;
  font-size: 20px;
}

<div
  class="css-5qt57y"
/>
`;
